docs/StorageManagement.md

```diff
@@ -0,0 +1,281 @@
+
+- IT admin should be able to express config policies based on storage dynamic attributes (e.g., cost) as well as storage properties such as type, geography and others. 
+
+- Optimizer needs to ensure that the allocated storage type matches the connection that the module uses.
+
+- The selected storage should not necessarily match the source dataset connection in case of copying an existing asset.
+
+- [Future enhancement] The data user should be able to request a specific storage type, or to specify some of the connection properties (e.g., bucket name) inside FybrikApplication. 
+
+- The data user should be able to leave a choice of a storage type to Fybrik and organization policies.  
+
+## Goals 
+
+- Provide modules with a connection object for writing data
+
+- Share organization storage to store temporary or persistent data while hiding the details (credentials, server URLs, service endpoints, etc.) from the data user. This means that data users will store data in organization accounts created by IT administrators.  
+
+- Govern the use of the shared storage for the given workload according to the compliance, capacity and other factors.
+
+- Optimize the shared storage for the given workload (by cost, latency to the workload)
+
+- Manage storage life cycle of the shared storage (e.g., delete temporary data after not being used, delete empty buckets, etc.) 
+
+# High Level Design
+
+## Taxonomy and structures
+
+It is crucial that the modules and the storage allocation component have a base common connection structure.  Otherwise, the components will not work together correctly when deployed.  Thus, we propose the following:
+
+- **Connection** is defined in **base** taxonomy within Fybrik repository, as it is done today:
+
+```
+// +kubebuilder:pruning:PreserveUnknownFields
+// Name of the connection to the data source
+type Connection struct {
+	// Name of the connection to the data source
+	Name                 ConnectionType   `json:"name"`
+	AdditionalProperties serde.Properties `json:"-"`
+}
+```
+
+- Fybrik defines **connection taxonomy layer** with schema definition for supported connections (in pkg/taxonomy/layers). Quickstart deploys Fybrik using this layer. Users can modify the connection layer when deploying Fybrik.
+
+In **Phase1** we define a connection taxonomy layer for all connection types that are supported today by open-source modules, such as `s3`, `db2`, `kafka`, `arrow-flight`. (Revisit taxonomy definition used by Airbyte module.)
+
+See [connection taxonomy](https://github.com/fybrik/fybrik/blob/master/samples/taxonomy/example/catalog/connection.yaml) for an example of a connection taxonomy layer.
+
+- Selection of modules is done based on connection `name`.
+
+In **Phase2** we add an optional `Category` field to `Connection`. 
+A module yaml will specify as a connection type either name or category. Optimizer will do the matching.
```

Comment:
No, it can't specify both in a single entry - there is only one field. There is no question of precedence - either the module supports the connection, or it is not.